File: shared-atlas-dart/lib/src/interfaces/presence_join_payload.dart
class PresenceJoinPayload {
  final String userId;
  final String username;
  final String? avatar;
  final String color;

  PresenceJoinPayload({
    required this.userId,
    required this.username,
    this.avatar,
    required this.color,
  });

  factory PresenceJoinPayload.fromJson(Map<String, dynamic> json) {
    return PresenceJoinPayload(
      userId: json['userId'] as String,
      username: json['username'] as String,
      avatar: json['avatar'] as String?,
      color: json['color'] as String,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'userId': this.userId,
      'username': this.username,
      'avatar': this.avatar,
      'color': this.color,
    };
  }
}

File: packages/shared-atlas-rust/src/interfaces/presence_join_payload.rs
use serde::{Serialize, Deserialize};

/// Event when a user joins the collaboration session.
/// 用户加入协作会话时的事件。
/// 
/// Event: `server.emit('presence:join', payload)`
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PresenceJoinPayload {
/// User ID / 用户ID
    pub user_id: String,
/// Username / 用户名
    pub username: String,
/// User avatar URL / 用户头像URL
    pub avatar: Option<String>,
/// Assigned color / 分配的颜色
    pub color: String,
}